frontend/src/components/App/UpperRightUI.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component, PropTypes } from 'react'
+
+import AccountMenu from './AccountMenu'
+import LoginForm from './LoginForm'
+import NotificationIcon from './NotificationIcon'
+
+class UpperRightUI extends Component {
+  static propTypes = {
+    currentUser: PropTypes.object,
+    signInPage: PropTypes.bool,
+    unreadNotificationsCount: PropTypes.number
+  }
+
+  static contextTypes = {
+    location: PropTypes.object
+  }
```

Comment:
contextTypes are special properties that can then be accessed from any child component under `this.context`. There are only very specific circumstances where they're recommended
https://facebook.github.io/react/docs/context.html